Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -56,17 +56,75 @@ public struct Reducer<State, Action, Environment> {
     Self { _, _, _ in .none }
   }
 
-  /// Combines many reducers into a single one by running each one on the state, and merging
+  /// Combines many reducers into a single one by running each one on state in order, and merging
   /// all of the effects.
   ///
+  /// It is important to consider the order of reducers passed to this function, especially when
+  /// combining a child domain with a parent domain that can modify the child domain by listening to
+  /// its actions.
```

Comment:
I'm thinking we need to explicitly say that `combine(a, b) != combine(b, a)` before diving into the `optional` example. 

```suggestion
  /// It is important to note that the order of combining reducers matter. Combining `reducerA` with 
  /// `reducerB` is not necessarily the same as combing `reducerB` with `reducerA`.
```

And should we also mention this also applies to `forEach`?

Reply:
Added some more language around `forEach`.